File: 31_10_2022/cpp/#3_daycon.cpp
#include <iostream>

int main() {
	freopen("./daycon.inp", "r", stdin);
	freopen("./daycon.out", "w+", stdout);

	int num_of_nums, m;
	std::cin >> num_of_nums;

	int nums[num_of_nums] = {};
	for (int i = 0; i < num_of_nums; i++) {
		std::cin >> nums[i];
	}
	std::cin >> m;

	int max, start = 0;
	for (int i = 0; i <= num_of_nums-m; i++) {
		int sum = 0;
		for (int j = i; j < i + m; j++) {
			sum += nums[j];
		}
		if (sum > max) {
			max = sum;
			start = i;
		}
	}

	for (int i = start; i < start + m; i++) {
		std::cout << nums[i] << ' ';
	}
}

File: 31_10_2022/py/#3_daycon.py
finp = open('./daycon.inp')

finp.readline()
nums = tuple(int(i) for i in finp.readline().split())
m = int(finp.readline())

max_, out = 0, []

for i in range(len(nums)-m+1):
	s = sum(nums[i:i+m])
	if s > max_:
		max_ = s
		out = nums[i:i+m]

print(*out, file=open('./daycon.out', 'w+'))